app/Exceptions/Handler.php

```diff
@@ -49,18 +51,43 @@ public function report(Exception $exception)
      */
     public function render($request, Exception $exception)
     {
-        if (! $request->expectsJson()) {
-            return parent::render($request, $exception);
+        if (strpos($request->getUri(), '/api/v') !== false) {
+            if ($exception instanceof NotFoundHttpException) {
+                return response()->json([
+                    'message' => 'Requested route is invalid.',
+                ], 400);
+            } elseif ($exception instanceof AuthenticationException) {
+                return response()->json([
+                    'message' => 'Unable to authenticate request, check if your access token is correct.',
+                ], 401);
+            } elseif ($exception instanceof HttpException) {
+                if ($exception->getMessage() == 'Too Many Attempts.') {
```

Comment:
So possibly the exception code is not always unique for each class... 

However we still can avoid using the messages, since there are exceptions for each concrete case, check them here:
https://api.symfony.com/3.4/Symfony/Component/HttpKernel/Exception.html

I suppose by adding more switch statements we will solve the problem.

A few of these exceptions extend from some more generic ones like ` TooManyRequestsHttpException` extends from `HttpException` that's why the error is caught there too.

Reply:
I've been having a hard time reproducing the TooManyRequestsHttpException, I eventually found this:
https://github.com/laravel/framework/issues/13623
It's mentioned that Laravel does not use TooManyRequestsHttpException but I can't find anything else that confirms this other than my experience trying to catch it.
How should I proceed? Is there another alternative other than using getMessage()?

Reply:
I wasn't aware that they don't use the `TooManyRequestsHttpException`.

So as mentioned in the Laravel thread, the two possible solutions would be to create a custom middleware to handle the ThrottleRequests or to use the `$exception->getStatusCode() == 429` as you said.

To keep everything in the same scope I think it would suffice to use the status code approach since we don't need to override too much stuff.